keps/sig-node/4569-cgroup-v1-deprecation/README.md

```diff
@@ -0,0 +1,685 @@
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Deprecate the support for cgroup v1 in Kubernetes, aligning with the industry's move towards cgroup v2 as the default for Linux kernel resource management and isolation.
+
+## Motivation
+
+cgroup v2 offers significant improvements over cgroup v1 in terms of feature set, uniformity of interface, and scalability. With major Linux distributions and container runtimes supporting cgroup v2, it is timely for Kubernetes to deprecate cgroup v1 support to streamline maintenance and encourage the adoption of newer, more secure, and more performant technologies.
+
+
+### Goals
+
+- Deprecation of cgroup v1 support in Kubernetes.
+- Provide clear migration paths for users running Kubernetes on systems using cgroup v1.
+- Ensure test coverage
+
+### Non-Goals
+
+Removing cgroup v1 support immediately. The deprecation process will be gradual, allowing users ample time to migrate.
+
+## Proposal
+
+The proposal outlines a plan to deprecate cgroup v1 support in Kubernetes, encouraging the community and users to transition to cgroup v2.
+
+### Risks and Mitigations
+
+The main risk involves potential disruptions to users who have not migrated to cgroup v2. Mitigations include providing clear documentation on the migration process, offering community support for common issues encountered during migration, and maintaining cgroup v1 support for a predefined deprecation period to allow for a smoother transition.
```

Comment:
Is the plan for the k8s community to work on the documentation, or to offer support? This is listed as one of the goals but I don't see much work outlined in the KEP.

Reply:
Good point. A comms plan would help; we can look at the removal of dockership for inspiration.